clash-cores/test/Test/Cores/Xilinx/DcFifo.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Test.Cores.Xilinx.DcFifo ( tests ) where
+
+import qualified Prelude as P
+import qualified Data.List as L
+import Data.Maybe (catMaybes, fromMaybe)
+import Control.Monad (replicateM)
+
+import Clash.Explicit.Prelude
+import Clash.Cores.Xilinx.DcFifo
+import Clash.Netlist.Util (orNothing)
+
+
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Hedgehog (MonadGen, (===), property, forAll, Property)
+import qualified Hedgehog as H
+import Test.Tasty.Hedgehog (testPropertyNamed)
+import Test.Tasty (testGroup, TestTree)
+
+tests :: TestTree
+tests = testGroup "FIFO tests"
+  [ testPropertyNamed
+    "FIFO doesn't lose any data with small stalls"
+    "prop_noloss"
```

Comment:
Isn't it more concise to just test the smallest capacity? (set `depth` to 4). That would cover all behavior we care about!

Reply:
That might be true because 16 is already pretty large and only powers of two are supported, eliminating a bunch of interesting edge cases.

I would still add at least one bigger one. You want it to cover a bunch of different patterns at different read and write clock ratios, then 16 is getting a bit cramped.

Reply:
I'd also like to add something in between, to still get backpressure but with a bit more different scenarios at a somewhat large clock ratio, meaning one side is pretty slow in responding to changes from the other side.

Reply:
I believe this is done.